common-auth/SECURITY.md

```diff
@@ -0,0 +1,65 @@
+# Running registry and streamline web-services securely
+
+This module is intended to be used by registry and streamline web-services so that they can enable http client authentication via SPNEGO. Currently supported 
+authentication mechanism is Kerberos. The code for this module has been borrowed from hadoop-auth(2.7.3) module in hadoop project and changed a little. The 
+reasons for doing so are to avoid having a dependency on hadoop-auth module which brings in some other modules, avoid conflicts with other versions of 
+hadoop-auth module and having more control over the changes needed in future. Some text for this document has been borrowed from SECURITY.md of Apache Storm  
+
+By default, registry and streamline web-services are running without authentication enabled and anyone can access the web-services from ui/client as far as they
+know the url and can access the web-server from the client machine. To enable authentication with the client, webservice needs to add a servlet filter from 
+this module. The webservice module will need to declare a dependency on this module. One way of adding a servlet filter in code is as follows. 
+
+```java
+List<ServletFilterConfiguration> servletFilterConfigurations = registryConfiguration.getServletFilters();
+if (servletFilterConfigurations != null && !servletFilterConfigurations.isEmpty()) {
+    for (ServletFilterConfiguration servletFilterConfiguration: servletFilterConfigurations) {
+        try {
+            FilterRegistration.Dynamic dynamic = environment.servlets().addFilter(servletFilterConfiguration.getClassName(), (Class<? extends Filter>)
+            Class.forName(servletFilterConfiguration.getClassName()));
+            dynamic.setInitParameters(servletFilterConfiguration.getParams());
+            dynamic.addMappingForUrlPatterns(EnumSet.allOf(DispatcherType.class), true, "/*");
+        } catch (Exception e) {
+            LOG.error("Error registering servlet filter {}", servletFilterConfiguration);
+            throw new RuntimeException(e);
+        }
+    }
+}
+```
+
+ServletFilterConfiguration is a java object representing any servlet filter used in schema registry. However the general idea is that one needs to
+add com.hortonworks.registries.auth.server.AuthenticationFilter for enabling authentication 
+
+The configuration for that filter in yaml is represented in params property as follows.
```

Comment:
oh, I know now what you mean. I would write it as:

The filter configuration is passed using the params property in they YAML file, as follows: